Microsoft.Toolkit.Uwp.UI.Controls/RangeSelector/RangeSelector.cs

```diff
@@ -606,6 +619,92 @@ private static void IsTouchOptimizedChangedCallback(DependencyObject d, Dependen
             rangeSelector.ArrangeForTouch();
         }
 
+        /// <summary>
+        /// Gets or sets the step value for range selector.
+        /// </summary>
+        /// <value>
+        /// The value for step.
+        /// </value>
+        public double StepValue
+        {
+            get
+            {
+                return (double)GetValue(StepValueProperty);
+            }
+
+            set
+            {
+                SetValue(StepValueProperty, value);
+            }
+        }
+
+        private static void StepValueChangedCallback(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var rangeSelector = d as RangeSelector;
+            if (rangeSelector == null)
+            {
+                return;
+            }
+
+            if (rangeSelector.StepValue != 0)
+            {
+                rangeSelector.ResetStepValues();
+            }
+        }
+
+        private void ResetStepValues()
+        {
+            if (StepValue != 0)
+            {
+                if (stepValues == null)
+                {
+                    stepValues = new List<double>();
+                }
+                else
+                {
+                    stepValues.Clear();
+                }
+
+                double newStep = Minimum + StepValue;
+                while (newStep < Maximum)
```

Comment:
Can't this be done using math? MoveToStepValue could just use the rangeValue and play some math with it (mod and subtraction). A while loop seems unnecessary.

Reply:
Agreed: if the only point in having the `stepValues` list is for the `MoveToStepValue` method, then it should be possible to do the same in that method just with simple math!